### DISCUSSION TITLE
Adding an "onStart" callback to note scheduling

### DISCUSSION THREAD

**Discussion by hylu-dev** [danigb/smplr#28] [Ideas]
Hi, awesome library! I came from using the original soundfont-player some years back and I'm really loving this successor.

As for my suggestion. I'm looking to hook up the note scheduling to a piano app I'm working on. Right now, I can use the onEnded callback to trigger key releases on the piano but it'd nice to also have an onStart callback to trigger keypresses as well.

**Comment by danigb** [comment]
Added in v0.11.0

**Reply by drscottlobo** [reply]
Awesome!! So happy you decided to make this work! Can't wait to check it out!

**Reply by hylu-dev** [reply]
Same here! Will try changing out my code with the new callback.

**Reply by drscottlobo** [reply]
I just tried after installing new version, I called with player.start() but it doesn't seem to fire yet, did this work on your end @hylu-dev?

Here's my code:
```

player.start({
			note: "C4",
			duration: 1,
			onEnded: () => {
				console.log("I ended!");
				// will be called after 1 second
			},
			onStart: () => {
				console.log("started!");
			},
		}); 
```

**Reply by hylu-dev** [reply]
Didn't get a chance to test it out in my project to confirm but I quickly tried in the sandbox.

https://codesandbox.io/s/young-monad-forked-lw8pyj?file=/src/index.js

Looks to be logging okay.

**Reply by drscottlobo** [reply]
It certainly does, I'm going to try a reinstall...did you install from npm?

**Reply by drscottlobo** [reply]
The only real difference I can see is that you loaded SplendidGrandPiano and I'm loading Soundfont, I wonder if somehow the Soundfont class doesn't have the Event yet?

**Reply by hylu-dev** [reply]
The sandbox should be pulling from npm. Nice catch on the instrument! Going offline for a bit, hopefully it’s a simple fix there.

**Comment by drscottlobo** [comment]
Hi @hylu-dev, I just forked that codepen you added, but instead loaded a Soundfont marimba instead, it plays fine, but no onStart is getting logged. Could the Soundfont class be missing the onStart event? @danigb is that possible? 

https://codesandbox.io/s/young-monad-forked-gy3jwq?file=/src/index.js
